agent/util-scripts/pbench-tool-meister

```diff
@@ -374,15 +317,18 @@ class ToolMeister(object):
         self.state = "start"
         self._state_trans = {
             "start": {"next": "stop", "action": self.start_tools},
-            "stop": {"next": "send", "action": self.stop_tools},
-            "send": {"next": "start", "action": self.send_tools},
+            "stop": {"next": "start", "action": self.stop_tools},
+            "send": {"next": None, "action": self.send_tools},
         }
         self._valid_states = frozenset(["start", "stop", "send", "terminate"])
         for key in self._state_trans.keys():
             assert (
                 key in self._valid_states
             ), f"INTERNAL ERROR: invalid state transition entry, '{key}'"
-            assert self._state_trans[key]["next"] in self._valid_states, (
+            assert (
+                self._state_trans[key]["next"] is None
```

Comment:
And this looks a bit odd; and also somewhat weakens the assert since it'll pass if any state has a 'next' of None.

Again, you've moved the "send" out of the state machine, which is why it doesn't follow the rules. I'm not sure I really like this change.

Reply:
I agree, this is not a great change, and it highlights that this is really not a state machine.

The state machine should define states: "running tools" and "idle", and then we transition between states via "start" and "stop" messages.  The "send" and "terminate" messages should not be part of the state machine.

Reply:
Fixed.